std/algorithm/searching.d

```diff
@@ -1788,6 +1815,50 @@ if (isRandomAccessRange!R1 && hasLength!R1 && hasSlicing!R1 && isBidirectionalRa
     {
         return haystack[haystack.length .. haystack.length];
     }
+    // Optimization in case the ranges are both SortedRanges.
+    // Binary search can be used to find the first occurence
+    // of the first element of the needle in haystack.
+    // When it is found O(walklength(needle)) steps are performed.
+    // 8829 enhancement
+    import std.range;
+    static if(is(typeof(haystack) == typeof(needle))
+            && is(typeof(haystack) : SortedRange!TT, TT)
+            && pred == "a == b")
+    {
+        auto needleFirstElem = needle[0];
+        auto partitions      = haystack.trisect(needleFirstElem);
+        auto firstElemLen    = partitions[1].length;
+        int count            = 0;
+
+        if(firstElemLen == 0)
```

Comment:
Replace most of the code below with a call to `mismatch`: https://dlang.org/library/std/algorithm/comparison/mismatch.html